### PR TITLE
Emission Dials

### PR DESCRIPTION
**Pull request by chrisjgf**
## Changelog:  
- Add Emission Dials page
  - [x] Emissions Epoch view
  - [x] System Dials Table
  - [x] User Dials Table
  - [x] Sidebar
  - [x] Form validation
  - [x] Submission

### CODE REVIEW

File: libs/base/src/lib/components/core/ExplorerLink.tsx [hunk 1]
```diff
@@ -3,18 +3,20 @@ import React, { FC } from 'react'
...
 
 export const ExplorerLink: FC<{
   data: string
   type?: 'transaction' | 'account' | 'address' | 'token'
   truncate?: boolean
   showData?: boolean
   className?: string
-}> = ({ children, type = 'address', data, showData, truncate = true, className }) => {
-  const getExplorerUrl = useGetExplorerUrl()
+  chainId?: ChainIds
```
Comment (JamesLefrere) [inline libs/base/src/lib/components/core/ExplorerLink.tsx:14]: You can now ask for an explorer link on a chain you're not connected to (useful for dials table)

File: libs/artifacts/src/graphql/emissions/query.graphql [hunk 2]
```diff
@@ -0,0 +1,87 @@
+fragment DialPreferences on Preference {
```
Comment (JamesLefrere) [inline libs/artifacts/src/graphql/emissions/query.graphql:1]: It's possible we could have more fragments here, but it's not a huge set of queries so maybe not worth it

File: libs/dumb-components/src/lib/Slider.tsx [hunk 3]
```diff
@@ -46,20 +57,32 @@ const StyledSlider = styled(ReactSlider)`
...
+      ref={ref}
+      disabled={disabled}
+      className={className}
       marks={markRange}
       renderTrack={renderTrack}
-      renderThumb={renderThumb}
+      renderThumb={(!disabled && renderThumb) || undefined}
       renderMark={renderMark}
-      onChange={v => onChange(v as number)}
+      onChange={v => onChange?.(v as number)}
```
Comment (rafaelugolini) [inline libs/dumb-components/src/lib/Slider.tsx:85]: this looks odd, no? what's the case where `onChange` won't exist?

File: apps/governance/src/app/pages/Dials/context/EpochContext.tsx [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+const EpochUpdater: FC = () => {
+  const clients = useApolloClients()
+  const [emissionsData] = useEmissionsData()
+  const [weekNumber] = useEpochWeekNumber()
+  const [, setEpochData] = useEpochData()
+  const [, setHoveredDialId] = useHoveredDialId()
+  const [, setSelectedDialId] = useSelectedDialId()
+
+  // TODO pagination
+  const skip = 0
```
Comment (JamesLefrere) [inline apps/governance/src/app/pages/Dials/context/EpochContext.tsx:46]: We will need to come back to this at some point, but we need more voters :)

File: apps/governance/src/app/pages/Dials/context/EpochContext.tsx [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+
+const [useEpochData, EpochDataProvider] = createStateContext<Epoch | undefined>(undefined)
+
+const [useEpochWeekNumber, EpochWeekNumberProvider] = createStateContext<number | undefined>(undefined)
+
+const useBlockNumberForWeekNumber = (): number | undefined => {
+  const clients = useApolloClients()
+  const [weekNumber] = useEpochWeekNumber()
+
+  // TODO revert
```
Comment (JamesLefrere) [inline apps/governance/src/app/pages/Dials/context/EpochContext.tsx:20]: Could remove this, but it's useful for testing Ropsten

**Review by JamesLefrere — COMMENTED** [review]
Can't see any howlers

File: libs/dumb-components/src/lib/Slider.tsx (hunk 3 above)
Comment (JamesLefrere) [reply libs/dumb-components/src/lib/Slider.tsx:85]: I also don't see that case, I'll make it required

**Review by rafaelugolini — APPROVED** [review]

**Pull request merged**